### PR TITLE
Require FSx Storage Capacity

### PR DESCRIPTION
**Pull request by sean-smith**
Signed-off-by: Sean Smith <seaam@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/pcluster/commands.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ def _check_for_updates(pcluster_config):
         utils.check_if_latest_version()
 
 
-def create(args):  # noqa: C901 FIXME!!!
+def create(args):  # noqa: C901 FIXME!!!w
```
Comment (rexcsn) [inline cli/pcluster/commands.py]: typo?

File: cli/pcluster/config/validators.py [hunk 2]
```diff
@@ -136,6 +136,9 @@ def fsx_validator(section_key, section_label, pcluster_config):
     if fsx_export_path and not fsx_import_path:
         errors.append("When specifying 'export_path', the 'import_path' option must be specified")
 
+    if not fsx_section.get_param_value("storage_capacity") and not fsx_section.get_param_value("fsx_fs_id"):
+        errors.append("When specifying 'fsx' section, the 'storage_capacity' option must be specified")
```
Comment (rexcsn) [inline cli/pcluster/config/validators.py]: Should we also check for the specific storage capacity as outlined by FSx: `For Lustre file systems, valid values are 1,200, 2,400, 3,600, then continuing in increments of 3600 GiB.`

We should at least provide this info in the error message so the user does not have to look it up again

File: cli/tests/pcluster/config/test_validators.py [hunk 3]
```diff
@@ -484,16 +484,20 @@ def test_raid_validators(mocker, section_dict, expected_message):
...
-        ({"shared_dir": "/NONE"}, "/NONE cannot be used as a shared directory"),
-        ({"shared_dir": "/fsx"}, None),
+        ({"export_path": "test", "import_path": "test", "storage_capacity": 1200}, None),
+        (
+            {"export_path": "test", "storage_capacity": 1200},
+            "When specifying 'export_path', the 'import_path' option must be specified",
+        ),
+        ({"shared_dir": "NONE", "storage_capacity": 1200}, "NONE cannot be used as a shared directory"),
+        ({"shared_dir": "/NONE", "storage_capacity": 1200}, "/NONE cannot be used as a shared directory"),
+        ({"shared_dir": "/fsx"}, "the 'storage_capacity' option must be specified"),
```
Comment (rexcsn) [inline cli/tests/pcluster/config/test_validators.py]: Probably need to change this or add more tests according to comment above.

File: cli/pcluster/commands.py (hunk 1 above)
Comment (sean-smith) [reply cli/pcluster/commands.py]: yep

File: cli/pcluster/config/validators.py (hunk 2 above)
Comment (sean-smith) [reply cli/pcluster/config/validators.py]: That's done here https://github.com/aws/aws-parallelcluster/blob/develop/cli/pcluster/config/validators.py#L194

File: cli/tests/pcluster/config/test_validators.py (hunk 3 above)
Comment (sean-smith) [reply cli/tests/pcluster/config/test_validators.py]: That's done here https://github.com/aws/aws-parallelcluster/blob/develop/cli/tests/pcluster/config/test_validators.py#L714

**Comment by enrico-usai** [comment]
@sean-smith could you describe in the comment of the PR why this change is necessary? 
Has anything changed in FSx side or were we missing this already required parameter?

In future please add this kind of description in the commit message too.

**Comment by sean-smith** [comment]
@enrico-usai Previously the stack failed CloudFormation create with a very opaque error message "Filesystem failed to create". This was despite our docs saying the parameter was optional. It's only optional if the customer provides a `fsx_fs_id` for an existing filesystem.

This patch fixes that behaviour by validating `storage_capacity` is provided when no `fsx_fs_id` is given.

**Pull request merged**